lib/api-sync.js

```diff
@@ -61,7 +61,7 @@ function removeUpdatesInRequest(updatesInDb, updatesInRequest) {
 function processSyncAPI(datasetId, params, readDatasetClient, cb) {
   var queryParams = params.query_params || {};
   var metaData = params.meta_data || {};
-  var cuid = syncUtil.getCuid(params);
+  var cuid = syncUtil.getCuid(params, syncConfig.cuidGenerator);
```

Comment:
[Optional] I will code this into more object oriented aproach. 
`syncConfig.cuidGenerator.getCuid();`

We need to setup default generator to config if it's missing.